### ISSUE TITLE
Docs should say that hi/lo accept nulls

### ISSUE THREAD

**Issue by neothemachine** [scijs/ndarray#23]
Since lo()/hi() accept nulls this should be added to the Readme as it is quite convenient.

**Comment by rreusser** [comment]
Can you elaborate on the use case? I've used it to just just slice a range in one particular dimension without specifying the bounds of the others.

**Comment by neothemachine** [comment]
Yes exactly, that's what I do as well, slicing across one or more dimensions and leaving others untouched (where I use null then instead of giving explicit full bounds).

**Comment by rreusser** [comment]
See: https://github.com/scijs/ndarray-concat-rows/blob/master/index.js#L72

I'm updating the docs for another ticket so will add a note on this now.

**Comment by neothemachine** [comment]
What about the example? I don't see any null's.

Am 07.01.2016 um 21:49 schrieb Ricky Reusser:

**Comment by rreusser** [comment]
Ah, sorry. Yeah, seemed (unless I'm wrong) that as long as it's undefined or null (omitted as an argument = undefined), lo doesn't increase the lower bound and hi doesn't decrease the upper bound in the unspecified dimensions.

**Comment by rreusser** [comment]
In other words, I used that to grab a section of the ndarray along the first dimension only by leaving the others implicitly unspecified. [Concatenating cols](https://github.com/scijs/ndarray-concat-cols/blob/master/index.js#L111-L116) was a little more complicated because I had to construct the array explicitly.

(Side note: I'm realizing as I clean some of this up that really more and more of the code should use dynamic code generation to keep this generic n-dimensional programming from actually having to operate in an abstract number of dimensions. With code generation, you just generate the code for the actual number of dimensions for the given input…)

But anyway, AFAIK, either null or implicitly undefined both works.

**Comment by neothemachine** [comment]
Right, ok.

About dynamic code generation, as long as it actually has a measureable performance benefit and the code is used enough times that it matters I'm all for it.
